cmd/version/version.go

```diff
@@ -64,3 +68,62 @@ func displayVersion(cmd *cobra.Command) {
 		Detail:  true,
 	})
 }
+
+func GetLatestVersion() (string, error) {
+	client := &http.Client{
+		CheckRedirect: func(req *http.Request, via []*http.Request) error { // no redirect
+			return http.ErrUseLastResponse
+		},
+	}
+	resp, err := client.Get("https://github.com/cisco-open/fsoc/releases/latest")
+	if err != nil {
+		return "", err
+	}
+	split := strings.Split(resp.Header.Get("Location"), "/")
+	if len(split) < 1 {
+		return "", fmt.Errorf("version request did not return a version")
+	}
+	return split[len(split)-1], nil
+}
+
+func CheckForUpdate(versionChannel chan *semver.Version) {
+	defer func() {
+		r := recover()
+		if r != nil {
+			log.Warnf("Error occurred while checking version, continuing")
+		}
+	}()
+	log.Infof("Checking for newer version of FSOC")
+	newestVersion, err := GetLatestVersion()
+	log.Infof("Latest version available: %s", newestVersion)
+	if err != nil {
+		log.Warnf(err.Error())
+	}
+	newestVersionSemVar, err := semver.NewVersion(newestVersion) // This panics, so we need to be ready to recover
+	if err != nil {
+		log.Warnf(err.Error())
+	}
+	versionChannel <- newestVersionSemVar
+}
+
+func CompareAndLogVersions(newestVersionSemVar *semver.Version) {
+	currentVersion := GetVersion()
+	currentVersionSemVer := ConvertVerToSemVar(currentVersion)
+	newerVersionAvailable := currentVersionSemVer.Compare(newestVersionSemVar) == -1
+	var debugFields = log.Fields{"newerVersionAvailable": newerVersionAvailable, "oldVersion": currentVersionSemVer.String(), "newVersion": newestVersionSemVar.String()}
+
+	if newerVersionAvailable {
+		log.WithFields(debugFields).Warnf("There is a newer version of FSOC available, please upgrade from version %s to version %s", currentVersionSemVer.String(), newestVersionSemVar.String())
```

Comment:
Since we already include the versions in the fields, I don't think we need to include them in the message as well. Just say "There is a newer version of fsoc available, please upgrade soon"